File: windows/src/ConnectorControl.Core/State/RemoteAuthKind.cs
namespace ConnectorControl.Core.State;

/// <summary>The four ways the Remote form can authenticate, in picker order.</summary>
public enum RemoteAuthKind
{
    Automatic,
    Bearer,
    Header,
    OAuthClient,
}

public static class RemoteAuthKindExtensions
{
    /// <summary>The picker's label for this kind.</summary>
    public static string Title(this RemoteAuthKind kind) => kind switch
    {
        RemoteAuthKind.Automatic => "Automatic (OAuth / none)",
        RemoteAuthKind.Bearer => "Bearer token",
        RemoteAuthKind.Header => "Custom header",
        RemoteAuthKind.OAuthClient => "OAuth client ID/secret",
        _ => throw new ArgumentOutOfRangeException(nameof(kind)),
    };
}

File: Sources/ConnectorControlState/RemoteAuthKind.swift
/// The four ways the Remote form can authenticate an `npx mcp-remote`
/// invocation, in picker order.
public enum RemoteAuthKind: String, CaseIterable, Sendable {
    case automatic, bearer, header, oauthClient

    public var title: String {
        switch self {
        case .automatic: return "Automatic (OAuth / none)"
        case .bearer: return "Bearer token"
        case .header: return "Custom header"
        case .oauthClient: return "OAuth client ID/secret"
        }
    }
}
